Grafana UI: Use Eslint no-restricted-imports in tests and stories

jackw opened the pull request:
<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Prevent grafana/ui tests and stories importing from packages the source code shouldn't import from.

**Why do we need this feature?**

To prevent implicit circular dependencies between files in our codebase. NX in particular doesn't like this as it does an analysis of the imports in code.

![image](https://github.com/user-attachments/assets/49ee2239-16ac-423c-a7c3-369bb4c7fed3)


**Who is this feature for?**

Grafana maintainers

**Which issue(s) does this PR fix?**:

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

jackw committed:
style(eslint): remove ignore from grafana/ui-overrides

jackw committed:
style(grafana-ui): update imports for tests

jackw committed:
style(grafana-ui): update imports for stories

jackw committed:
chore(betterer): update results file

tomratcliffe commented on packages/grafana-ui/src/components/PanelChrome/PanelChrome.story.tsx:
```diff
@@ -5,13 +5,19 @@ import { CSSProperties, useState, ReactNode } from 'react';
+import { PanelChromeProps } from './PanelChrome';
 import mdx from './PanelChrome.mdx';
 
+import { PanelChrome } from '.';
```
(Is it clearer to do `./index`? I guess it doesn't matter though 🤷 )

tomratcliffe committed:
Fix useTheme import

tomratcliffe committed:
Change deprecated Horizontal/Vertical group usages

tomratcliffe commented:
I pushed 2 changes - fixed a `useTheme2` import that wasn't working in storybook, and removed the 2 deprecated vertical/horizontal group usages and updated bettere

jackw commented:
> I pushed 2 changes - fixed a `useTheme2` import that wasn't working in storybook, and removed the deprecated vertical/horizontal group usages and updated betterer

@tomratcliffe thanks so much for helping me with this. 🙏

tomratcliffe reviewed (approved):
LGTM! 🚀
